exchangerates/kraken.go

```diff
@@ -0,0 +1,116 @@
+package exchangerates
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	// KrakenPairSiacoinUSD is the ID of SC/USD pair in Kraken
+	KrakenPairSiacoinUSD = "SCUSD"
+	// KrakenPairSiacoinEUR is the ID of SC/EUR pair in Kraken
+	KrakenPairSiacoinEUR = "SCEUR"
+)
+
+type krakenAPI struct {
+	client http.Client
+}
+
+type krakenPriceResponse struct {
+	Error  []any `json:"error"`
+	Result map[string]struct {
+		A []string `json:"a"`
+		B []string `json:"b"`
+		C []string `json:"c"`
+		V []string `json:"v"`
+		P []string `json:"p"`
+		T []int    `json:"t"`
+		L []string `json:"l"`
+		H []string `json:"h"`
+		O string   `json:"o"`
+	} `json:"result"`
+}
+
+func newKrakenAPI() *krakenAPI {
+	return &krakenAPI{}
+}
+
+// See https://docs.kraken.com/api/docs/rest-api/get-ticker-information
+func (k *krakenAPI) ticker(pair string) (float64, error) {
+	pair = strings.ToUpper(pair)
+	response, err := k.client.Get("https://api.kraken.com/0/public/Ticker?pair=" + url.PathEscape(pair))
```

Comment:
Would be a bit cleaner to interrupt the http requests with a context on shutdown that you pass in